internal/auth/oauth2.go

```diff
@@ -420,13 +426,23 @@ func (as *AuthenticationService) VerifyClientCredentials(ctx context.Context, cl
 	return nil
 }
 
+// sessionCrossOriginSafe sets parameters on the session that allow its use in cross-origin requests.
+// Options are not saved to the database so this must be called whenever a session cookie will be returned to a client.
+//
+// Note browsers require cookies with the same-site policy as 'none' to additionally have the secure flag set.
+func sessionCrossOriginSafe(session *sessions.Session, secure bool) *sessions.Session {
+	session.Options.Secure = secure                  // Ensures only sent with HTTPS
+	session.Options.HttpOnly = true                  // Don't allow Javascript to modify cookie
+	session.Options.SameSite = http.SameSiteNoneMode // Allow cross-origin requests via Javascript
```

Comment:
i think it's possible to set a domain for cookies, like `.canonical`. I can imagine in a production environment having the dashboard and jimm under the same subdomain.
In this way we are more secure.

Reply:
The domain automatically gets set to the domain that set the cookie. [Doc source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value)
![image](https://github.com/user-attachments/assets/4a53e785-2264-4ffe-8add-56332819aa8a)